src/typechecker.erl

```diff
@@ -407,16 +418,17 @@ compat_ty({type, _, map, Assocs1}, {type, _, map, Assocs2}, Seen, Env) ->
                                     %% if that's not the case, let's throw now.
                                     length(MandatoryAssocs1) == 0 andalso throw(nomatch),
                                     case lists:foldl(fun
-                                                         (_Assoc1, {Seen1, Cs1}) -> {Seen1, Cs1};
-                                                         (Assoc1, nomatch) ->
+                                                         (Assoc1, {no, match}) ->
```

Comment:
Constraint solver limitation? Perhaps add a comment about it here too?

Reply:
Exactly 👍